File: Data Structures & Algorithms/duplicate-integer/submission-1.cpp
#include <unordered_map>
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        unordered_map<int, int> umap;
        for(int i =0; i <nums.size();i++){
            if (umap.count(nums[i]) == 0){
                umap[nums[i]]=1;
            }
            else{
                return true;
            }
        }
        return false;
    }
};

File: Data Structures & Algorithms/first-missing-positive/submission-1.py
class Solution:
    def firstMissingPositive(self, nums: List[int]) -> int:
        x=set()
        res=1
        for n in nums:
            x.add(n)
        for i in range(1,len(nums)+1):
            if i in x:
                res+=1
                continue
            else:
                return res
    
                
        return len(nums) + 1
